main.cpp

```diff
@@ -549,5 +612,61 @@ int main()
 
     divider();
 
+    std::cout << "The apply() function applies the current value of the current object to itself.\n\n";
+
+    FloatType ftA(4.5f);
+
+    std::cout << "FtA is currently: " << static_cast<float>(ftA) << std::endl;
+    
+    ftA.apply( [&]( float &a ) -> FloatType&
```

Comment:
- [ ] Don't capture more than you need to!  You are capturing **everything** in `main()` by reference.